src/notebook.c

```diff
@@ -65,6 +66,8 @@ static gboolean switch_in_progress = FALSE;
 static GtkWidget *switch_dialog = NULL;
 static GtkWidget *switch_dialog_label = NULL;
 
+static gboolean doc_saves_to_new_file = FALSE;
```

Comment:
Do you mean to have something like `g_signal_connect(geany_object, "document-open", G_CALLBACK(on_document_open), &doc_saves_to_new_file);`?  Should we really do that?